Check if buf is valid inside vim.schedule

Akianonymus opened the pull request:
When a buffer get deleted on escape, then that buffer will not be valid and will throw an error

Error executing vim.schedule lua callback: .../aki/.local/share/nvim/lazy/houdini.nvim/lua/houdini.lua:113: Invalid buffer id: 3348 stack traceback:
	[C]: in function 'nvim_buf_get_lines'
	.../aki/.local/share/nvim/lazy/houdini.nvim/lua/houdini.lua:113: in function <.../aki/.local/share/nvim/lazy/houdini.nvim/lua/houdini.lua:112>

Akianonymus committed:
fix: check if buf is valid inside vim.schedule

TheBlob42 commented:
Hey thank you for your pull request :slightly_smiling_face: 

I have a few requests before merging:
- please adopt your commit message to the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification  
(see the other commits in the repository for some inspiration)
- change your indentation to spaces and match it to the surrounding code, currently it seems pretty off
- you can remove the outer parentheses from your if statement:  
`if not vim.api.nvim_buf_is_valid(buf) then` 

And because I'm interested: Do you have any specific use case where the buffer gets deleted on escaping?

Akianonymus commented:
Changed the commit message and fixed the indentation.

I use https://github.com/VonHeikemen/searchbox.nvim, it shows a popup which closes on esc. Here is a video demonstrating the bug.

---

https://user-images.githubusercontent.com/32216346/220568035-e66b4e58-ad27-43c2-b17b-eff399c676c5.mp4

TheBlob42 commented:
Perfect :+1: 

Thanks for the use case example. Makes perfect sense to me now :slightly_smiling_face:
